src/LoriotAzureFunctions/Route/RouteFunction.cs

```diff
@@ -76,74 +77,95 @@ public ReturnMessage()
         }
 
         [FunctionName("RouteFunction")]
-        [return: EventHub("outputEventHubMessage", Connection = "EVENT_HUB_ROUTER_OUTPUT")]
-        public async static Task<string> Run([EventHubTrigger("%IotHubName%", Connection = "EVENT_HUB_ROUTER_INPUT", ConsumerGroup = "router")]EventData myEventHubMessage,
+        public async static Task Run([EventHubTrigger("%IOT_HUB_NAME%", Connection = "EVENT_HUB_ROUTER_INPUT", ConsumerGroup = "router")]EventData[] myEventHubMessageInput,
+            [EventHub("outputEventHubMessage", Connection = "EVENT_HUB_ROUTER_OUTPUT") ]IAsyncCollector<String> output,
               TraceWriter log)
         {
-            //section to build up the metadata section
-            var deviceId = GetDeviceId(myEventHubMessage);
-            dynamic metadataMessageSection;
-            if (localCache.Contains(deviceId))
-            {
-                metadataMessageSection = localCache[deviceId];
-            }
-            else
-            {
-                metadataMessageSection = await GetTags(System.Environment.GetEnvironmentVariable("IOT_HUB_OWNER_CONNECTION_STRING"), deviceId);
-                localCache.Add(deviceId, metadataMessageSection, policy);
-            }
+            foreach (var myEventHubMessage in myEventHubMessageInput) {
+                //section to build up the metadata section
+                var deviceId = GetDeviceId(myEventHubMessage);
+                dynamic metadataMessageSection;
+                //retry logic to avoid the initial message rush to be declined by the IoT hub.
+                int retryCount = 0;
+                for (; ; )
+                {
+                    try
+                    {
+                        if (localCache.Contains(deviceId))
+                        {
+                            metadataMessageSection = localCache[deviceId];
+                        }
+                        else
+                        {
+                            metadataMessageSection = await GetTags(System.Environment.GetEnvironmentVariable("IOT_HUB_OWNER_CONNECTION_STRING"), deviceId);
+                            localCache.Add(deviceId, metadataMessageSection, policy);
+                        }
+                        break;
+                    }catch (Exception ex)
+                    {
+                        retryCount++;
+                        if (retryCount > 5)
+                            throw new Exception("Could not connect with the IoT Hub device manager");
+                        await Task.Delay(5000);
```

Comment:
The default delay could be something like 1 second that should be enough to avoid the original problem.